src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+        {
+            tooltip.accept(Helpers.translatable("tfc.jade.input_stacks", bloomery.getInputStacks().size()));
+            tooltip.accept(Helpers.translatable("tfc.jade.catalyst_stacks", bloomery.getCatalystStacks().size()));
+            if (state.getValue(BloomeryBlock.LIT))
+            {
+                final long ticksLeft = bloomery.getRemainingTicks();
+                if (ticksLeft > 0)
+                {
+                    final BloomeryRecipe recipe = bloomery.getCachedRecipe();
+                    if (recipe != null)
+                    {
+                        tooltip.accept(Helpers.translatable("tfc.jade.hours_remaining", String.valueOf(Mth.ceil((float) ticksLeft / ICalendar.TICKS_IN_HOUR))));
+                        tooltip.accept(Helpers.translatable("tfc.jade.creating", recipe.getResultItem().getHoverName()));
+                    }
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip BLOOM = (level, state, entity, tooltip) -> {
+        if (entity instanceof BloomBlockEntity bloom)
+        {
+            final ItemStack item = bloom.getItem();
+
+            ItemStack displayItem = item.copy();
+            displayItem.setCount(bloom.getCount());
+            itemWithCount(tooltip, displayItem);
+
+            final List<Component> text = new ArrayList<>();
+            item.getCapability(HeatCapability.CAPABILITY).ifPresent(cap -> cap.addTooltipInfo(item, text));
+            text.forEach(tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip CHARCOAL_FORGE = (level, state, entity, tooltip) -> {
+        if (entity instanceof CharcoalForgeBlockEntity forge)
+        {
+            heat(tooltip, forge.getTemperature());
+        }
+    };
+
+    public static final BlockEntityTooltip COMPOSTER = (level, state, entity, tooltip) -> {
+        if (state.getBlock() instanceof TFCComposterBlock block && entity instanceof ComposterBlockEntity composter)
+        {
+            if (composter.isRotten())
+            {
+                tooltip.accept(Helpers.translatable("tfc.composter.rotten"));
+            }
+            else
+            {
+                hoeOverlay(level, block, entity, tooltip);
+                if (!composter.isReady())
+                {
+                    final String percent = Helpers.formatPercentage(Math.min(0.99f, (float) composter.getTicksSinceUpdate() / composter.getReadyTicks()) * 100);
+                    tooltip.accept(Helpers.translatable("tfc.jade.progress").append(percent));
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip CROP = (level, state, entity, tooltip) -> {
+        if (entity instanceof CropBlockEntity crop && state.getBlock() instanceof CropBlock block)
+        {
+            hoeOverlay(level, block, entity, tooltip);
+
+            tooltip.accept(Helpers.translatable("tfc.jade.yield").append(Helpers.literal(String.format("%.2f", crop.getYield()))));
```

Comment:
```suggestion
            tooltip.accept(Helpers.translatable("tfc.jade.yield", String.format("%.2f", crop.getYield())));
```
Will need to modify the translation key.